src/js/utils/lifecycle-callbacks.js

```diff
@@ -17,4 +17,15 @@ export default class LifecycleCallbacksMixin {
     }
     this.callbackQueues[queueName].push(callback);
   }
+  addCallbackOnce(queueName, callback) {
+    if (!queueName) { throw new Error('Must pass queue name to addCallbackOnce'); }
+    if (!this.callbackQueues[queueName]) {
+      this.callbackQueues[queueName] = [];
+    }
+    let index = this.callbackQueues[queueName].indexOf(callback);
+    if (index !== -1) {
+      this.callbackQueues[queueName].splice(index, 1);
+    }
+    this.callbackQueues[queueName].push(callback);
```

Comment:
Is there a reason to force the callback to always be at the end of the queue like this? If not would be simpler to:
```
if (index === -1) {
  this.callbackQueues[queueName].push(callback);
}
```